### PR TITLE
allow passing multiple columns arguments to select and select!

### PR DESCRIPTION
**Pull request by bkamins**
Fixes #1955.

### CODE REVIEW

**Comment by bkamins** [comment]
See the tests for examples how to move columns to the front and to the end.

**Comment by bkamins** [comment]
also deprecate `permutecols!`

**Comment by bkamins** [comment]
@nalimilan - I have updated the documentation for this PR. Could you please have a look at it before it is merged? Thank you!

File: src/dataframe/dataframe.jl [hunk 1]
```diff
@@ -768,10 +768,16 @@ end
 """
     select!(df::DataFrame, inds)
 
-Mutate `df` in place to retain only columns specified by `inds` and return it.
+Mutate `df` in place to retain only columns specified by `inds...` and return it.
 
-Argument `inds` can be any index that is allowed for column indexing of
-a `DataFrame` provided that the columns requested to be removed are unique.
+Arguments passed as `inds...` can be any index that is allowed for column indexing
+provided that the columns requested in eachof them are unique and present in `df`.
```
Comment (nalimilan) [inline src/dataframe/dataframe.jl]:
```suggestion
provided that the columns requested in each of them are unique and present in `df`.
```

File: src/dataframe/dataframe.jl [hunk 2]
```diff
@@ -768,10 +768,16 @@ end
 """
     select!(df::DataFrame, inds)
```
Comment (nalimilan) [inline src/dataframe/dataframe.jl]:
```suggestion
    select!(df::DataFrame, inds...)
```

**Comment by bkamins** [comment]
@nalimilan - thank you!

**Comment by bkamins** [comment]
The test failure on nightly is unrelated to this PR.

I think https://github.com/JuliaLang/julia/commit/13a8fbcb659680b5a28486a6e6291025a2d166b0 caused the test on nightly to fail. I will re-implement the failing tests sometime next week in a separate PR.

CC @quinnj as it might impact also other packages

**Pull request merged**